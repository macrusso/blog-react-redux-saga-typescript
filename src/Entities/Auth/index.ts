import * as authConstants from './authConstants';
import * as authActions from './authActions';
import * as authSagas from './authSagas';
import authReducer, { ICommentsState } from './authReducer';

export { authSagas, ICommentsState, authActions, authReducer, authConstants };
