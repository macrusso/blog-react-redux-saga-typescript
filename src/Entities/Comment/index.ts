import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
import * as commentSagas from './commentSagas';
import commentReducer, { ICommentsState } from './commentReducer';
import ICommentPartial from './ICommentPartial';
import IComment from './IComment';

export {
  IComment,
  commentSagas,
  ICommentsState,
  commentActions,
  commentReducer,
  ICommentPartial,
  commentSelectors,
  commentConstants,
};
