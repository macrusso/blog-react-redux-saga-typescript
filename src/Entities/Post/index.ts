import * as postConstants from "./postConstants";
import * as postSelectors from "./postSelectors";
import * as postActions from "./postActions";
import * as postSagas from "./postSagas";
import postReducer from "./postReducer";
import Post from "../../Frontend/Post/PostListItem";
import IPostPartial from "./IPostPartial";
import IPost from "./IPost";
import IPostsState from "./IPostsState";

export { Post, IPost, postSagas, postActions, IPostsState, postReducer, IPostPartial, postSelectors, postConstants };
