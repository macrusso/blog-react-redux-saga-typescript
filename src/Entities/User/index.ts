import * as userConstants from './userConstants';
import * as userSelectors from './userSelectors';
import * as userActions from './userActions';
import * as userSagas from './userSagas';
import userReducer, { IUsersState } from './userReducer';
import IUserAuth from './IUserAuth';
import IUserPartial from './IUserPartial';
import IUser from './IUser';

export {
  IUser,
  IUserAuth,
  userSagas,
  userActions,
  IUsersState,
  userReducer,
  IUserPartial,
  userSelectors,
  userConstants,
};
