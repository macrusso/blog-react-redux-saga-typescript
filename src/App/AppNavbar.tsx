import React from "react";
import { Button, Typography, Toolbar, AppBar } from "@material-ui/core";
import { IUser } from "../Entities";

interface IAppNavbarProps {
  currentUser?: IUser;
  onLogoClick: () => void;
  onLoginClick: () => void;
  onLogoutClick: () => void;
  onRegisterClick: () => void;
}

const AppNavbar: React.SFC<IAppNavbarProps> = props => {
  const { currentUser, onLogoClick, onLoginClick, onLogoutClick, onRegisterClick } = props;
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography variant="h6" color="inherit" onClick={() => onLogoClick()}>
          Simple React Blog
        </Typography>
        {(currentUser && currentUser.token) || localStorage.token ? (
          <Button color="inherit" onClick={() => onLogoutClick()}>
            Logout
          </Button>
        ) : (
          <>
            <Button color="inherit" onClick={() => onRegisterClick()}>
              Register
            </Button>
            <Button color="inherit" onClick={() => onLoginClick()}>
              Login
            </Button>
          </>
        )}
      </Toolbar>
    </AppBar>
  );
};

export default AppNavbar;
