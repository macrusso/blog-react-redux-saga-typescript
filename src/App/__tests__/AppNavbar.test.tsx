import React from "react";
import { Button, Typography, Toolbar, AppBar } from "@material-ui/core";
import AppNavbar from "../AppNavbar";
import { shallow } from "enzyme";
import { IUser } from "../../Entities";
import { BrowserRouter } from "react-router-dom";

describe("AppNavbar", () => {
  test("Snapshot matches", () => {
    const user: IUser = {
      name: "string",
      email: "string",
      _id: "5bcc554259e0dbfda6ed1a55",
    };

    const wrapper = shallow(
      <BrowserRouter>
        <AppNavbar
          currentUser={user}
          onLogoClick={jest.fn()}
          onLoginClick={jest.fn()}
          onLogoutClick={jest.fn()}
          onRegisterClick={jest.fn()}
        />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
