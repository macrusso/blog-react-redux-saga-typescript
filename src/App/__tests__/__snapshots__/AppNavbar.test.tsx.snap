// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavbar Snapshot matches 1`] = `ShallowWrapper {}`;
